src/Toml/Parser/Value.hs

```diff
@@ -284,3 +285,10 @@ valueP = UBool    <$> boolP
      <|> UInteger <$> integerP
      <|> UText    <$> textP
      <|> UArray   <$> arrayP
+
+
+anyValueP :: Parser AnyValue
+anyValueP = valueP >>= \uval ->
+  case typeCheck uval of
+    Left err -> fail $ show err
```

Comment:
I feel like this function can be written in the following way:

```haskell
anyValueP = typeCheck <$> valueP >>= \case
    Left err -> ...
    Right -> ...
```

Personally, to me the priority of `<$>` and `>>=` is always clear and this code is anambiguous. But when you write something like `foo >>= x =<< bar` you should reconsider your coding style... Combination of `<$>` and `>>=` is totally valid.

Reply:
That's a good suggestion, but did I use `foo >>= x =<< bar` somewhere?

Reply:
> but did I use foo >>= x =<< bar somewhere?

No, you haven't 🙂 I saw it once previously. The idea here is that operators can be a mess but combination of `<$>` and `>>=` is actually quite idiomatic and fine 👍